src/xenia/kernel/xobject.h

```diff
@@ -128,10 +133,22 @@ class XObject {
   Memory* memory() const;
 
   Type type();
-  X_HANDLE handle() const;
+
+  // Returns the primary handle of this object.
+  X_HANDLE handle() const { return handles_[0]; }
+
+  // Returns all associated handles with this object.
+  std::vector<X_HANDLE> handles() const { return handles_; }
+  std::vector<X_HANDLE>& handles() { return handles_; }
+
   const std::string& name() const { return name_; }
   uint32_t guest_object() const { return guest_object_ptr_; }
 
+  // Has this object been created for use by the host?
+  // Host objects are persisted through reloads/etc.
+  bool host_object() const { return host_object_; }
```

Comment:
style tip: bools like this named as is_* (so `is_host_object`) - makes it easy to know what it is (instead of if you just saw `host_object()`, you may think it's actually a pointer to some host object or something)